docs/integration/aspnetcore.rst

```diff
@@ -399,31 +399,85 @@ In standard ASP.NET Core integration and generic hosted application support ther
 
     public class Program
     {
-      public static void Main(string[] args)
+      public static async Task Main(string[] args)
       {
+        // create the root-container and register global dependencies
+        var containerBuilder = new ContainerBuilder();
+
+        builder.RegisterType<SomeGlobalDependency>()
+          .As<ISomeGlobalDependency>()
+          .InstancePerLifetimeScope();
+
+        var container = containerBuilder.Build();
+
         // The UseServiceProviderFactory call attaches the
         // Autofac provider to the generic hosting mechanism.
-        var host = Host.CreateDefaultBuilder(args)
-            .UseServiceProviderFactory(new AutofacChildLifetimeScopeServiceProviderFactory())
+          var hostOne = Host
+            .CreateDefaultBuilder(args)
+            .UseServiceProviderFactory(new AutofacChildLifetimeScopeServiceProviderFactory(container))
```

Comment:
```suggestion
            .UseServiceProviderFactory(new AutofacChildLifetimeScopeServiceProviderFactory(container.BeginLifetimeScope("root-one")))
```

The idea here is that the "root" can be a child scope of the container, it doesn't have to be a container at all. I'll update the two examples with some suggestions and we should be able to call it good.